woocommerce/woocommerce-ios
# Edit Products: update product name edit UI to inline text field

Fixes #1957 

## Changes

- Added `UITextField` extension to apply headline styles
- Created `TextFieldTableViewCell` that contains a text field + xib
- Updated `ProductFormTableViewDataSource` and `ProductFormSection.PrimaryFieldRow` configuration to use the new `TextFieldTableViewCell`
- In `ProductFormViewController` and `ProductFormTableViewDataSource`, updated the name change handling. When the name changes, it still updates its `product` but it won't trigger a table view reload if name is the only change

## Testing

- Go to the Products tab
- Tap on a simple Product
- Edit the product name however you like. Try clearing the title to make sure the placeholder says "Title"
- Tap on any other row, change some data there, and tap "Done" --> it should go back to the Edit Product main screen, and the product name should remain the same as the last edited value
- Tap "Update" --> the product name should be updated remotely after it completes

## Example screenshots

Screencast:

![simulator](https://user-images.githubusercontent.com/1945542/77153877-98103a00-6ad5-11ea-885d-d6f006561d74.gif)

mode \ state | empty | non-empty
-- | -- | --
dark | ![Simulator Screen Shot - iPhone 11 - 2020-03-20 at 18 06 18](https://user-images.githubusercontent.com/1945542/77153932-b5450880-6ad5-11ea-9788-686ef8708d79.png) | ![Simulator Screen Shot - iPhone 11 - 2020-03-20 at 18 06 20](https://user-images.githubusercontent.com/1945542/77154451-ad399880-6ad6-11ea-8f30-ccf407e1d7d8.png)
light | ![Simulator Screen Shot - iPhone 11 - 2020-03-20 at 18 05 37](https://user-images.githubusercontent.com/1945542/77154470-b9bdf100-6ad6-11ea-8bc8-d3a002bcd635.png) | ![Simulator Screen Shot - iPhone 11 - 2020-03-20 at 18 05 40](https://user-images.githubusercontent.com/1945542/77154481-bdea0e80-6ad6-11ea-90e0-9fda0e513407.png)

 
Update release notes:
